XamarinCommunityToolkit/Helpers/WeakEventManagerService.shared.cs

```diff
@@ -0,0 +1,178 @@
+                    }
+                }
+                catch (TargetParameterCountException e)
+                {
+                    throw new InvalidHandleEventException("Parameter count mismatch. If invoking an `event Action` use `HandleEvent(string eventName)` or if invoking an `event Action<T>` use `HandleEvent(object eventArgs, string eventName)`instead.", e);
+                }
+            }
+        }
+
+        internal static void HandleEvent(in string eventName, in object actionEventArgs, in Dictionary<string, List<Subscription>> eventHandlers)
+        {
+            AddRemoveEvents(eventName, eventHandlers, out var toRaise);
+
+            for (var i = 0; i < toRaise.Count; i++)
+            {
+                try
+                {
+                    var tuple = toRaise[i];
+                    if (tuple.Item2.IsLightweightMethod())
+                    {
+                        var method = TryGetDynamicMethod(tuple.Item2);
+                        method?.Invoke(tuple.Item1, new[] { actionEventArgs });
+                    }
+                    else
+                    {
+                        tuple.Item2.Invoke(tuple.Item1, new[] { actionEventArgs });
+                    }
+                }
+                catch (TargetParameterCountException e)
+                {
+                    throw new InvalidHandleEventException("Parameter count mismatch. If invoking an `event EventHandler` use `HandleEvent(object sender, TEventArgs eventArgs, string eventName)` or if invoking an `event Action` use `HandleEvent(string eventName)`instead.", e);
+                }
+            }
+        }
+
+        internal static void HandleEvent(in string eventName, in Dictionary<string, List<Subscription>> eventHandlers)
+        {
+            AddRemoveEvents(eventName, eventHandlers, out var toRaise);
+
+            for (var i = 0; i < toRaise.Count; i++)
+            {
+                try
+                {
+                    var tuple = toRaise[i];
+                    if (tuple.Item2.IsLightweightMethod())
+                    {
+                        var method = TryGetDynamicMethod(tuple.Item2);
+                        method?.Invoke(tuple.Item1, null);
+                    }
+                    else
+                    {
+                        tuple.Item2.Invoke(tuple.Item1, null);
+                    }
+                }
+                catch (TargetParameterCountException e)
+                {
+                    throw new InvalidHandleEventException("Parameter count mismatch. If invoking an `event EventHandler` use `HandleEvent(object sender, TEventArgs eventArgs, string eventName)` or if invoking an `event Action<T>` use `HandleEvent(object eventArgs, string eventName)`instead.", e);
+                }
+            }
+        }
+
+        static void AddRemoveEvents(in string eventName, in Dictionary<string, List<Subscription>> eventHandlers, out List<Tuple<object, MethodInfo>> toRaise)
```

Comment:
Can you use `ValueTuple` instead of `Tuple`? If is possible to use `ValueTuple`, please name the `Item1` and `Item2` to better names.

Reply:
No problem! ✅